doc/development_process/code_flow.rst

```diff
@@ -0,0 +1,43 @@
+
+Code Flow and Branches
+######################
+
+Introduction
+************
+
```

Comment:
Zephyr consists of 3 types of branches:

- `master` which contains the latest state of development
- `topic-` topic branches that are used for shared development of a new feature
- `vx.y-branch` branches which track maintenance releases based on a major release